### ISSUE TITLE
Would it be possible to have it base on Arabizi instead?

### ISSUE THREAD

**Issue by sawtdakhili** [joehakimrahme/buckwalter.el#4]
As you know, [Arabizi](https://en.wikipedia.org/wiki/Arabic_Chat_Alphabet) is the most popular transliteration. Would it be difficult to implement it?

Something like [Yamli](https://www.yamli.com/arabic-keyboard/) but for emacs.

**Comment by joehakimrahme** [comment]
Arabizi (thanks, I didn't know it was called that) is considerably harder to implement. The main reason is that there's no **1 to 1 mapping between arabic letters and latin ones**. This means that there is no single "correct" way to transliterate something from one script to the next.

Also, I'm not an expert, but I get the feeling that most of the written Arabizi relates to spoken dialects and not literary Arabic script, making it even harder to translate. 

Take your username for example: Going purely by alphabet, it could be transliterated in multiple ways:

- سوت دكهِلِ
- ساوط ضكحلي

It's only by actually learning arabic vocabulary that one can make the educated guess that it's صوت داخلي. This requires knowledge of the language.

Yamli works by giving transliteration candidates for every word and letting the user pick one. This is much more work than what my package is meant to do.

Buckwalter is meant for the use case where you want to write arabic, you have already made up your mind about the spelling, and lack the arabic input keys on your keyboard. It's NOT meant to infer the spelling of a word based on approximate latin characters.


**TL;DR:** It's much harder than a simple 1:1 conversion, which is what my package does.
